### DISCUSSION TITLE
jwt-auth  customized header not working

### DISCUSSION THREAD

**Discussion by liujun69** [apache/apisix#7407] [Q&A]
According to https://apisix.apache.org/docs/apisix/plugins/jwt-auth/；I am already set customer setting, then set the jwt-auth plugin in route use {"header":"X-AccessToken", "disable": false}, and then i request 9080 port use header "X-AccessToken", it not working and response {"message":"Missing JWT token in request"}

**Comment by liujun69** [comment]
jwt-auth.lua:344: phase_func(): failed to fetch JWT token: JWT not found in cookie, client: 172.21.0.1 in error.log, why not fetch in header

**Reply by tokers** [reply]
Which APISIX version are you using?

**Reply by liujun69** [reply]
try it on 2.12.1 and 2.14.1, It doesn't work

**Reply by liujun69** [reply]
2022/07/08 01:47:57 [info] 43#43: *2200541 [lua] init.lua:398: http_access_phase(): matched route: {"clean_handlers":{},"key":"\/apisix\/routes\/411105500743074497","value":{"priority":0,"create_time":1654567573,"upstream_id":"412265531865825985","id":"411105500743074497","uri":"\/api\/**","name":"bss","update_time":1657174426,"methods":["GET","POST","PUT","DELETE","PATCH","HEAD","OPTIONS","CONNECT","TRACE"],"plugins":{"proxy-rewrite":{"regex_uri":["^\/api\/(.*)","\/$1"]},"jwt-auth":{"header":"X-AccessToken","disable":false},"ext-plugin-pre-req":{"disable":true,"allow_degradation":false,"conf":[{"value":"{\"url\":\"http:\/\/172.16.15.131:18080\/app\/auth\"}","name":"auth"}]}},"status":1},"has_domain":false,"orig_modifiedIndex":32752,"modifiedIndex":32752,"createdIndex":18974,"update_count":0}, client: 127.0.0.1, server: _, request: "GET /api/inner/dept/org HTTP/1.1", host: "127.0.0.1:9080"
2022/07/08 01:47:57 [info] 43#43: *2200541 [lua] jwt-auth.lua:344: phase_func(): failed to fetch JWT token: JWT not found in cookie, client: 127.0.0.1, server: _, request: "GET /api/inner/dept/org HTTP/1.1", host: "127.0.0.1:9080"

**Comment by liujun69** [comment]
when I check the jwt_auth.lua under /usr/local/apisix in docker:
local function fetch_jwt_token(ctx)                                                      
    local token = core.request.header(ctx, "authorization")                              
    if token then                                                                        
        local prefix = sub_str(token, 1, 7)                                              
        if prefix == 'Bearer ' or prefix == 'bearer ' then                               
            return sub_str(token, 8)                                                     
        end                                                                              
                                                                                         
        return token                                                                     
    end                            
and check the source code in github
local function fetch_jwt_token(conf, ctx)
    local token = core.request.header(ctx, conf.header)
    if token then
        local prefix = sub_str(token, 1, 7)
        if prefix == 'Bearer ' or prefix == 'bearer ' then
            return sub_str(token, 8)
        end

        return token
    end
Is there any initialization configuration missing？

**Comment by liujun69** [comment]
it work in the latest version

**Reply by tzssangglass** [reply]
this feature was added in v2.13.0, see: https://github.com/apache/apisix/pull/6561
